modules/apps/asset/asset-list-test/src/testIntegration/java/com/liferay/asset/list/internal/model/listener/test/GroupModelListenerTest.java

```diff
@@ -0,0 +1,127 @@
+import com.liferay.portal.kernel.service.GroupLocalServiceUtil;
+import com.liferay.portal.kernel.test.rule.AggregateTestRule;
+import com.liferay.portal.kernel.test.rule.DeleteAfterTestRun;
+import com.liferay.portal.kernel.test.util.GroupTestUtil;
+import com.liferay.portal.kernel.test.util.TestPropsValues;
+import com.liferay.portal.kernel.util.GetterUtil;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.portal.kernel.util.UnicodeProperties;
+import com.liferay.portal.kernel.util.UnicodePropertiesBuilder;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.segments.model.SegmentsEntry;
+import com.liferay.segments.test.util.SegmentsTestUtil;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Joshua Cords
+ */
+@RunWith(Arquillian.class)
+public class GroupModelListenerTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new LiferayIntegrationTestRule();
+
+	@Before
+	public void setUp() throws Exception {
+		_group = GroupTestUtil.addGroup();
+		_segmentsEntry = SegmentsTestUtil.addSegmentsEntry(
+			TestPropsValues.getGroupId());
+	}
+
+	@Test
+	public void testRemoveGroupReferenceFromTypeSettings() throws Exception {
+		AssetListEntry assetListEntry1 = _createAssetListEntry(
+			new long[] {_group.getGroupId()});
+
+		AssetListEntry assetListEntry2 = _createAssetListEntry(
+			new long[] {_group.getGroupId(), TestPropsValues.getGroupId()});
+
+		GroupLocalServiceUtil.deleteGroup(_group.getGroupId());
+
+		Assert.assertEquals(0, _getGroupId(assetListEntry1));
+
+		Assert.assertEquals(
+			TestPropsValues.getGroupId(), _getGroupId(assetListEntry2));
+	}
+
+	private AssetListEntry _createAssetListEntry(long[] groupIds)
+		throws Exception {
+
+		AssetListEntry assetListEntry = AssetListTestUtil.addAssetListEntry(
+			TestPropsValues.getGroupId(),
+			AssetListEntryTypeConstants.TYPE_DYNAMIC);
+
+		AssetListEntrySegmentsEntryRel assetListEntrySegmentsEntryRel =
+			AssetListTestUtil.addAssetListEntrySegmentsEntryRel(
+				TestPropsValues.getGroupId(), assetListEntry,
+				_segmentsEntry.getSegmentsEntryId());
+
+		UnicodeProperties unicodeProperties = UnicodePropertiesBuilder.create(
+			true
+		).put(
+			"groupIds", StringUtil.merge(groupIds)
+		).build();
+
+		assetListEntrySegmentsEntryRel.setTypeSettings(
+			unicodeProperties.toString());
+
+		_assetListEntrySegmentsEntryRelLocalService.
+			updateAssetListEntrySegmentsEntryRelTypeSettings(
+				assetListEntry.getAssetListEntryId(),
+				_segmentsEntry.getSegmentsEntryId(),
+				unicodeProperties.toString());
+
+		return assetListEntry;
+	}
+
+	private long _getGroupId(AssetListEntry assetListEntry1) throws Exception {
+		assetListEntry1 = _assetListEntryLocalService.getAssetListEntry(
```

Comment:
i dont think this needs to be named assetListEntry1
and maybe the method param can just be the assetListEntryId, then create a new AssetListEntry instead of reassigning the one you passed in